recordriver/driver.go

```diff
@@ -141,7 +141,19 @@ func (s *stmt) Query(_ []driver.Value) (driver.Rows, error) {
 	sess := s.session
 	sessions[sess].Queries = append(sessions[sess].Queries, s.query)
 	if resp, ok := sessions[sess].responses[s.query]; ok {
-		return resp, nil
+		var copiedResp Response
+		copiedResp.Cols = make([]string, len(resp.Cols))
+		for i := range resp.Cols {
+			copiedResp.Cols[i] = resp.Cols[i]
+		}
+		copiedResp.Data = make([][]driver.Value, len(resp.Data))
+		for i := range resp.Data {
+			copiedResp.Data[i] = make([]driver.Value, len(resp.Data[i]))
+			for j := range resp.Data[i] {
+				copiedResp.Data[i][i] = resp.Data[i][j]
+			}
+		}
+		return &copiedResp, nil
```

Comment:
Move into Clone/Copy response method. Feels more elegant 

Also, stick with the current short naming.